vsintegration/src/FSharp.Editor/CodeFix/AddMissingEqualsToTypeDefinition.fs

```diff
@@ -0,0 +1,62 @@
+﻿// Copyright (c) Microsoft Corporation.  All Rights Reserved.  See License.txt in the project root for license information.
+
+namespace Microsoft.VisualStudio.FSharp.Editor
+
+open System
+open System.Composition
+open System.Threading.Tasks
+
+open Microsoft.VisualStudio.FSharp.Editor.Logging
+
+open Microsoft.CodeAnalysis.Text
+open Microsoft.CodeAnalysis.CodeFixes
+
+[<ExportCodeFixProvider(FSharpConstants.FSharpLanguageName, Name = "AddMissingEqualsToTypeDefinition"); Shared>]
+type internal FSharpAddMissingEqualsToTypeDefinitionCodeFixProvider() =
+    inherit CodeFixProvider()
+
+    let fixableDiagnosticIds = set ["FS0010"; "FS3360"]
+
+    override _.FixableDiagnosticIds = Seq.toImmutableArray fixableDiagnosticIds
+
+    override _.RegisterCodeFixesAsync context : Task =
+        asyncMaybe {
+            let diagnostics =
+                context.Diagnostics
+                |> Seq.filter (fun x -> fixableDiagnosticIds |> Set.contains x.Id)
+                |> Seq.toImmutableArray
+
+            for diagnostic in diagnostics do
+                let message = diagnostic.GetMessage()
```

Comment:
Is this really the best we can do?

Reply:
The error number is already used. This just ensures only these errors that also propose the equals as a fix are used

Reply:
I will exclude FS0010. There can be specialized error messages for this and other cases and I can add separate quick fixes for those.